gcloud-java-bigquery/src/main/java/com/google/gcloud/spi/DefaultBigQueryRpc.java

```diff
@@ -417,4 +433,69 @@ public QueryResponse query(QueryRequest request) throws BigQueryException {
       throw translate(ex);
     }
   }
+
+  @Override
+  public String open(JobConfigurationLoad configuration) throws BigQueryException {
+    try {
+      Job loadJob = new Job().setConfiguration(new JobConfiguration().setLoad(configuration));
+      StringBuilder builder = new StringBuilder()
+          .append(BASE_RESUMABLE_URI)
+          .append(options.projectId())
+          .append("/jobs");
+      GenericUrl url = new GenericUrl(builder.toString());
+      url.set("uploadType", "resumable");
+      JsonFactory jsonFactory = bigquery.getJsonFactory();
+      HttpRequestFactory requestFactory = bigquery.getRequestFactory();
+      HttpRequest httpRequest =
+          requestFactory.buildPostRequest(url, new JsonHttpContent(jsonFactory, loadJob));
+      httpRequest.getHeaders().set("X-Upload-Content-Value", "application/octet-stream");
+      HttpResponse response = httpRequest.execute();
+      return response.getHeaders().getLocation();
+    } catch (IOException ex) {
+      throw translate(ex);
+    }
+  }
+
+  @Override
+  public void write(String uploadId, byte[] toWrite, int toWriteOffset, long destOffset, int length,
+      boolean last) throws BigQueryException {
+    try {
+      GenericUrl url = new GenericUrl(uploadId);
+      HttpRequest httpRequest = bigquery.getRequestFactory().buildPutRequest(url,
+          new ByteArrayContent(null, toWrite, toWriteOffset, length));
+      long limit = destOffset + length;
+      StringBuilder range = new StringBuilder("bytes ");
+      range.append(destOffset).append('-').append(limit - 1).append('/');
+      if (last) {
+        range.append(limit);
+      } else {
+        range.append('*');
+      }
+      httpRequest.getHeaders().setContentRange(range.toString());
+      int code;
+      String message;
+      IOException exception = null;
+      try {
+        HttpResponse response = httpRequest.execute();
+        code = response.getStatusCode();
+        message = response.getStatusMessage();
+      } catch (HttpResponseException ex) {
+        exception = ex;
+        code = ex.getStatusCode();
+        message = ex.getStatusMessage();
+      }
+      if (!last && code != HTTP_RESUME_INCOMPLETE
+          || last && !(code == HTTP_OK || code == HTTP_CREATED)) {
```

Comment:
I am missing how this can happen. The request for status must have content range of the form `Content-Range: bytes */<limit>`. If I am not mistaken this can never occur as we always set a range.